### PR TITLE
Add blast door crushing

### PR DESCRIPTION
**Pull request by Arrow768**
Revives Printers PR #4144

Adds blast door/shutter crushing. Does 40 damage for blast doors and 10 for shutters. Related thread: https://forums.aurorastation.org/viewtopic.php?f=18&t=10200

Edit: Tested it. Shutters / Blastdoors would have crushed windows which is somewhat of a issue considering the blastdoors on the bridge are located inside of the windows.
Removed window-crushing for now from airlocks aswell.
Wouldn't make sense if airlocks can crush windows, but blastdoors and shutters cant.

### CODE REVIEW

**Comment by Kaedwuff** [comment]
Holy crap, 40?  I don't know about that, you could kill someone in only a few seconds by slamming a blast door on them.  It's already kind of hard to kill unattended people by doorcrush as AI, this might make it incredibly easy to blastdoor smash security to death, especially when they're getting into the armory to arm up against you.

**Comment by Arrow768** [comment]
Yes, its stuns you, but it also moves you out of the door, so you are not stun locked in the door.
Unless the AI manages to trap you inside of the bridge entrence doors, it cant crush you to death.
(The ones where you have the airlock, then the blastdoor, then the airlock)

**Comment by Kaedwuff** [comment]
Well... as long as only the important people on station have an deathtrap of doom, I guess it's okay then.

File: code/game/machinery/doors/blast_door.dm [hunk 1]
```diff
@@ -147,6 +150,10 @@
 	if (src.operating || (stat & BROKEN || stat & NOPOWER))
 		return
 	force_close()
+	for(var/turf/turf in locs)
+		for(var/atom/movable/AM in turf)
+			if(AM.airlock_crush(damage))
+				take_damage(damage)
```
Comment (skull132) [inline code/game/machinery/doors/blast_door.dm]: This proc causes the door to take damage. A standard door has 300 health. This is, what, 8 ~~crushings~~ objects crushed before the door poops itself? I'd kinda expect them to be sturdier.

**Pull request merged**